File: solutions/csharp/1732-find-the-highest-altitude.cs
public class Solution {
    public int LargestAltitude(int[] gain) {
        int maxAltitude = 0; 
        int currentAltitude = 0; 

        foreach(int g in gain){ 
            currentAltitude += g; 
            if(currentAltitude > maxAltitude) 
                maxAltitude = currentAltitude; 
        } 
        return maxAltitude; 
                                                                                            
    }
}

File: solutions/rust/1732-find-the-highest-altitude.rs
impl Solution {
    pub fn largest_altitude(gain: Vec<i32>) -> i32 {
        let mut max_altitude = 0; 
        let mut current_altitude = 0; 

        for &g in gain.iter() { 
            current_altitude += g; 
            if current_altitude > max_altitude { 
                max_altitude = current_altitude; 
            } 
        } 
        max_altitude 
                                                                                                   
    }
}
